File: 3146.py
from collections import defaultdict

class Solution:
    def findPermutationDifference(self, s: str, t: str) -> int:
        mapper = defaultdict(lambda : 0)
        res = 0

        for index, letter in enumerate(s):
            mapper[letter] = index

        for index, letter in enumerate(t):
            res += abs(index - mapper[letter])
        
        return res

File: 3146.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {
        unordered_map<char, int> count;
        int res = 0;

        for(int i = 0; i < s.size(); i++) {
            count[s[i]] += i;
        }
        for(int i = 0; i < t.size(); i++) {
            count[t[i]] -= i;
            res += abs(count[t[i]]);
        }

        return res;        
    }
};